desci-server/src/redisClient.ts

```diff
@@ -96,3 +96,69 @@ export async function getOrCache<T>(key: string, fn: () => Promise<T>, ttl = DEF
     throw e;
   }
 }
+
+class SingleNodeLockService {
+  private isReady: boolean;
+  private MAX_LOCK_TIME = 60 * 60; // 1 hour
```

Comment:
Consider making this an envvar so we can tune it without rebuilding the service :ok_hand: